cmd/jaeger/internal/command.go

```diff
@@ -58,3 +51,16 @@ func Command() *cobra.Command {
 
 	return cmd
 }
+
+func handleConfigFlag(configFlag *pflag.Flag, args []string) error {
```

Comment:
not exactly what I meant, this still leaves a lot of code paths you cannot test. 
```shell
$ go test -cover ./cmd/jaeger/internal
ok  	github.com/jaegertracing/jaeger/cmd/jaeger/internal	0.841s	coverage: 75.7% of statements
```

Instead try this:
```go
	cmd.RunE = func(cmd *cobra.Command, args []string) error {
		return checkConfigAndRun(cmd, args, yamlAllInOne.ReadFile, otelRunE)
	}
// ...

// extracted for testability
func checkConfigAndRun(
	cmd *cobra.Command,
	args []string,
	getCfg func(name string) ([]byte, error),
	runE func(cmd *cobra.Command, args []string) error,
) error {
	configFlag := cmd.Flag("config")
	if !configFlag.Changed {
		log.Print("No '--config' flags detected, using default All-in-One configuration with memory storage.")
		log.Print("To customize All-in-One behavior, pass a proper configuration.")
		data, err := getCfg("all-in-one.yaml")
		if err != nil {
			return fmt.Errorf("cannot read embedded all-in-one configuration: %w", err)
		}
		configFlag.Value.Set("yaml:" + string(data))
	}
	return runE(cmd, args)
}

```